packages/swap/src/utils/getPoolQuote.ts

```diff
@@ -41,14 +48,23 @@ export default async function getPoolQuote({
   // https://linear.app/chainflip/issue/PRO-1370/include-boost-fees-in-quote-from-cf-swap-rate-v2
   if (boostFeeBps) {
     const boostFee = getPipAmountFromAmount(depositAmount, boostFeeBps);
-    includedFees.push(buildFee(srcAsset, 'BOOST', boostFee));
+    includedFees.push({
+      type: 'BOOST',
+      amount: boostFee,
+      ...internalAssetToRpcAsset[srcAsset],
+    } as const);
     cfRateInputAmount -= boostFee;
   }
 
   if (isVaultSwap) {
     excludeFees.push('IngressDepositChannel');
   }
 
+  if (isOnChain) {
+    excludeFees.push('Egress');
+    excludeFees.push('IngressDepositChannel');
```

Comment:
i think you also need to include the other ingress fee, otherwise the quote will include an ingress fee for vault swaps

Reply:
iirc bitcoin vault swaps have an egress fee because we stil need to fetch them

Reply:
so we should exclude this fee in most cases? we're not doing that currently

Reply:
https://github.com/chainflip-io/chainflip-backend/blob/d5babf740948eccdaa9e1a281649a67c254af8fd/state-chain/runtime/src/lib.rs#L1799

Reply:
:)